html/js/tree/Tree.js

```diff
@@ -173,10 +194,208 @@ GTE.TREE = (function (parentModule) {
+                if (nodes[i].player.id === playerId) {
+                    nodes[i].player = null;
+                }
+            }
+        }
+    };
+
+    /**
+    * Checks a colour is unique in the list of players
+    * @param  {String}  colour Hex code of the colour to check
+    * @return {Boolean} unique Returns false if the colour is not unique
+    */
+    Tree.prototype.checkColourIsUnique = function (colour) {
+        // Iterate over the list of players trying to find that specific colour
+        for (var i = 0; i < this.players.length; i++) {
+            if (colour === this.players[i].colour) {
+                // Return false if colour found
+                return false;
+            }
+        }
+        return true;
+    };
+
+    /**
+    * Hides tree's leaves
+    */
+    Tree.prototype.hideLeaves = function () {
+        var numberLeaves = this.numberLeaves();
+        for (var i = 0; i < numberLeaves; i++) {
+            this.leaves[i].hide();
+        }
+    };
+
+    /**
+    * Shows tree's leaves
+    */
+    Tree.prototype.showLeaves = function () {
+        var numberLeaves = this.numberLeaves();
+        for (var i = 0; i < numberLeaves; i++) {
+            this.leaves[i].show();
+        }
+    };
+
+    /**
+    * Returns the active player
+    * @return {Player} player Active Player
+    */
+    Tree.prototype.getActivePlayer = function () {
+        return this.players[GTE.tools.getActivePlayer()];
+    };
+
+    /**
+    * Gets all nodes in tree
+    * @return {List} listOfNodes List of tree's nodes
+    */
+    Tree.prototype.getAllNodes = function () {
+        var listOfNodes = [];
+        this.recursiveGetAllNodes(this.root, listOfNodes);
+        return listOfNodes;
+    };
+
+    /**
+    * Recursive function that gets all nodes in tree
+    * Stopping criteria: that the current node is a leaf
+    * Recursive expansion: to all of the node's children
+    * @param {Node} node        Node to expand through
+    * @param {List} listOfNodes List where nodes should be added and that will
+    *                           be returned by the non recursive function that
+    *                           calls this function
+    */
+    Tree.prototype.recursiveGetAllNodes = function (node, listOfNodes) {
+        for (var i = 0; i < node.children.length; i++) {
+            this.recursiveGetAllNodes(node.children[i], listOfNodes);
+        }
+        listOfNodes.push(node);
+    };
+
+    /**
+    * Get all player's nodes accross the tree
+    * @param {Number} playerId Id of the player
+    * @return {List} playerNodes List of nodes that belong to that player
+    */
+    Tree.prototype.getPlayerNodes = function (playerId) {
+        var allNodes = this.getAllNodes();
+        var playerNodes = [];
+        for (var i = 0; i < allNodes.length; i++) {
+            if (allNodes[i].player !== undefined &&
+                allNodes[i].player.id === playerId) {
```

Comment:
Bug: This can throw an uncaught typeerror, similarly to the situation in the previous comment

Steps to reproduce: 
1. Create a tree
2. add a new player (green)
3. Assign red player to one node
4. Assign green player to other node
5. Remove green player
6. Try to change the name of the red node assigned in step 3

To solve it, just check if allNodes[i].player !== undefined && **allNodes[i].player !== null** && ...